### ISSUE TITLE
Stacked Null Safety causing Type issues

### ISSUE THREAD

**Issue by FilledStacks** [Stacked-Org/stacked#283]
I'm getting this issue from flutter.

```dart
type '(BuildContext, NonReactiveViewModel, Widget?) => Scaffold' is not a subtype of type '(BuildContext, ChangeNotifier, Widget?) => Widget'
```

This is in the latest provider nullsafety package.

NonReactiveViewModel 's inheritance tree
- ChangeNotifier
    - BaseViewModel
      - NonReactiveViewModel

The code looks like this

View code
```dart
class NonReactiveView extends StatelessWidget {
  const NonReactiveView({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return ViewModelBuilder<NonReactiveViewModel>.nonReactive(
      builder: (context, model, child) => Scaffold(
        floatingActionButton: FloatingActionButton(
          onPressed: () {
            model.updateTitle();
          },
        ),
        body: Center(
          child: Text(model.title),
        ),
      ),
      viewModelBuilder: () => NonReactiveViewModel(),
    );
  }
}
```

Full error

```

════════ Exception caught by widgets library ═══════════════════════════════════
The following _TypeError was thrown building ViewModelBuilder<NonReactiveViewModel>(dirty, state: _ViewModelBuilderState<NonReactiveViewModel>#e6888):
type '(BuildContext, NonReactiveViewModel, Widget?) => Scaffold' is not a subtype of type '(BuildContext, ChangeNotifier, Widget?) => Widget'

The relevant error-causing widget was
ViewModelBuilder<NonReactiveViewModel>
lib\…\nonreactive\nonreactive_view.dart:11
When the exception was thrown, this was the stack
#0      _ViewModelBuilderState.build
package:stacked/…/state_management/view_model_builder.dart:148
#1      StatefulElement.build
package:flutter/…/widgets/framework.dart:4612
#2      ComponentElement.performRebuild
package:flutter/…/widgets/framework.dart:4495
#3      StatefulElement.performRebuild
package:flutter/…/widgets/framework.dart:4667
#4      Element.rebuild
package:flutter/…/widgets/framework.dart:4189
...
════════════════════════════════════════════════════════════════════════════════
```

**Comment by FilledStacks** [comment]
When I cast the model to a ChangeNotifier it works :/ 

```dart
class NonReactiveView extends StatelessWidget {
  const NonReactiveView({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return ViewModelBuilder<NonReactiveViewModel>.nonReactive(
      builder: (context, ChangeNotifier model, child) => Scaffold(
        floatingActionButton: FloatingActionButton(
          onPressed: () {
            // model.updateTitle();
          },
        ),
        body: Center(
          // child: Text(model.title),
        ),
      ),
      viewModelBuilder: () => NonReactiveViewModel(),
    );
  }
}
```

**Comment by a43501** [comment]
Error: type '(HomeViewModel) => dynamic' is not a subtype of type '((ChangeNotifier) => dynamic)?'
 after update to stacked: ^2.0.0-nullsafety.2. Please help. Thank you!

#0 _ViewModelBuilderState._createViewModel
package:stacked/…/state_management/view_model_builder.dart:113

#1 _ViewModelBuilderState.initState
package:stacked/…/state_management/view_model_builder.dart:93

`
          return ViewModelBuilder<HomeViewModel>.reactive(
            viewModelBuilder: () => HomeViewModel(),
            onModelReady: (model) => model.getData(),
            builder: (context, model, child) => Scaffold(
               body: Center(
               // child: Text(model.title),
               ),
            )
          )
`

**Comment by vptcnt** [comment]
Same error

```dart
class HomeView extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return ViewModelBuilder<HomeViewModel>.reactive(
        viewModelBuilder: () => HomeViewModel(),
        onModelReady: (viewModel) {
          // Do something once your viewModel is initialized
        },
        builder:
            (BuildContext context, HomeViewModel viewModel, Widget? child) {
          return ScreenTypeLayout(
            mobile: _HomeMobile(viewModel),
            desktop: _HomeDesktop(viewModel),
            tablet: _HomeTablet(viewModel),
          );
        });
  }
}
```
and
```dart
class HomeViewModel extends BaseViewModel {}
```
and
```dart
part of home_view;

class _HomeMobile extends StatelessWidget {
  final HomeViewModel viewModel;

  _HomeMobile (this.viewModel);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: Text('HomeMobile'),
      ),
    );
  }
}
```

**Comment by FilledStacks** [comment]
this should be fixed in stacked 2.0.0
